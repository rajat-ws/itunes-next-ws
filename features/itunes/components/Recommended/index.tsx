/**
 *
 * Recommended
 *
 */

import React from "react";
import { useRouter } from "next/router";
import { ClickableTags } from "@common";
import styled from "styled-components";

interface RecommendedProps {
  recommendations?: Recommendation[];
}

export interface Recommendation {
  trackId: number;
  trackName: string;
}

const StyledRecommendedContainer = styled.div`
  && {
    display: flex;
    gap: 2rem;
    justify-content: center;
  }
`;

const Recommended: React.FC<RecommendedProps> = ({ recommendations }) => {
  const router = useRouter();

  return (
    <StyledRecommendedContainer data-testid="recommended">
      {recommendations?.length &&
        recommendations.map(({ trackId, trackName }) => (
          <div
            data-testid="recommendTag"
            key={trackId}
            onClick={() => router.push(`/trackDetails/${trackId}`)}
          >
            <ClickableTags>{trackName} ðŸŽ¶ </ClickableTags>
          </div>
        ))}
    </StyledRecommendedContainer>
  );
};

export default Recommended;
