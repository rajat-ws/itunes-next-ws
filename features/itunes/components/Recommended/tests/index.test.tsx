/**
 *
 * Tests for Recommended
 *
 */

import React from "react";
import { fireEvent, render } from "@utils/testUtils";
import Recommended from "../index";

describe("<Recommended />", () => {
  const props = {
    recommendations: [
      {
        trackId: 12345,
        trackName: "Baarishein",
      },
    ],
  };

  it("should render and match the snapshot", () => {
    const { baseElement } = render(<Recommended {...props} />);
    expect(baseElement).toMatchSnapshot();
  });

  it("should contain 1 Recommended component", () => {
    const { getAllByTestId } = render(<Recommended {...props} />);
    expect(getAllByTestId("recommended").length).toBe(1);
  });

  it("should check for recommend-tag inside the component", () => {
    const { getByTestId } = render(<Recommended {...props} />);
    expect(getByTestId("recommendTag")).toBeInTheDocument();
    fireEvent.click(getByTestId("recommendTag"));
  });
});
