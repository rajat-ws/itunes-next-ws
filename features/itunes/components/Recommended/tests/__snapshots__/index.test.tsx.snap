// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Recommended /> should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="Recommended__StyledRecommendedContainer-sc-1a3frhj-0 bFbAXf"
      data-testid="recommended"
    >
      <div
        data-testid="recommendTag"
      >
        <span
          class="ant-tag styled__ClickableTags-sc-15gkald-2 kagfWg"
        >
          Baarishein
           ðŸŽ¶ 
        </span>
      </div>
    </div>
  </div>
</body>
`;
